### PR TITLE
Area search

### PR DESCRIPTION
**Pull request by CocoisBuggy**
I've generalized some of the code for adding new collections. Everything is importing to typescript without errors, I'm not sure how to properly test it though...

### CODE REVIEW

**Review by vnugent — CHANGES_REQUESTED** [review]
Thanks for this!   

In current version we accumulate the chunk array to max `chunkSize` and them push to Typesense **as** we loop through the Mongo result set.  Therefore at any given time there are only at most 5K (chunkSize) documents in memory.  

In the PR we add the entire Mongo collection to `chunks` array, which can be very large (~125K climb documents).

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
     "refresh-db": "./refresh-db.sh",
     "seed-usa": "yarn build && node build/db/import/usa/USADay0Seed.js",
     "init-db": "yarn build && node build/db/utils/jobs/UpdateAllRunner.js",
-    "update-search": "node build/db/export/Typesense/Typesense.js"
+    "update-search": "tsc ; node build/db/export/Typesense/Typesense.js"
```
Comment (vnugent) [inline package.json]: can we have a new and separate job?  Ex:  `yarn update-search-climbs` and `yarn update-search-areas`?  This way we can test and run each job independently without impacting one another.

**Review by vnugent — CHANGES_REQUESTED** [review]

**Comment by CocoisBuggy** [comment]
> Thanks for this!
> 
> In current version we accumulate the chunk array to max `chunkSize` and them push to Typesense **as** we loop through the Mongo result set. Therefore at any given time there are only at most 5K (chunkSize) documents in memory.
> 
> In the PR we add the entire Mongo collection to `chunks` array, which can be very large (~125K climb documents).

Ah, of course. I'll refactor for better memory use

File: package.json (hunk 1 above)
Comment (CocoisBuggy) [reply package.json]: Sure, I'll add that in

**Comment by vnugent** [comment]
@all-contributors add @CocoisBuggy for code and ideas

**Pull request merged**